elasticreindexer/process/reindexerMultiWrites.go

```diff
@@ -20,29 +20,21 @@ type reindexerMultiWrite struct {
 	blockChainStartTime  int64
 	enabled              bool
 
-	reindexerClient *reindexer
+	reindexerClient ReindexerHandler
 }
 
-func NewReindexerMultiWrite(cfg *config.GeneralConfig) (*reindexerMultiWrite, error) {
-	if cfg.Indexers.Input.URL == "" {
-		return nil, errors.New("empty url for the input cluster")
-	}
-	if cfg.Indexers.Output.URL == "" {
-		return nil, errors.New("empty url for the output cluster")
-	}
-
-	ri, err := CreateReindexer(cfg)
-	if err != nil {
-		return nil, err
+func NewReindexerMultiWrite(reindexer ReindexerHandler, cfg config.IndicesConfig) (*reindexerMultiWrite, error) {
+	if reindexer == nil {
+		return nil, errors.New("nil ReindexerHandler")
 	}
 
 	return &reindexerMultiWrite{
-		reindexerClient:      ri,
-		indicesNoTimestamp:   cfg.Indexers.IndicesConfig.Indices,
-		indicesWithTimestamp: cfg.Indexers.IndicesConfig.WithTimestamp.IndicesWithTimestamp,
-		numParallelWrite:     cfg.Indexers.IndicesConfig.WithTimestamp.NumParallelWrites,
-		blockChainStartTime:  cfg.Indexers.IndicesConfig.WithTimestamp.BlockchainStartTime,
-		enabled:              cfg.Indexers.IndicesConfig.WithTimestamp.Enabled,
+		reindexerClient:      reindexer,
+		indicesNoTimestamp:   cfg.Indices,
+		indicesWithTimestamp: cfg.WithTimestamp.IndicesWithTimestamp,
```

Comment:
for this we still don`t have any check.

Reply:
what should happen if NumParallelWrites is set 0 ?
what should happen if BlockchainStartTime is 1?

Reply:
1. If `NumParallelWrites` is less than 2 it will create a single interval.
2. For **BlockchainStartTime** added an extra check to be greater than zero 
If is 1 will calculate intervals based on this value